### PR TITLE
Fix issues in the objectstorage based on JuiceFS use cases.

### PR DESCRIPTION
**Pull request by XDTD**
## Description

"Fixing issues in the object storage based on JuiceFS use cases. Mainly includes:

1. Route path of createbucket.
2. Unable to properly pass objectkey when calling copyobject.
3. Oss GetObjectMeta return not found error.
4. getObjectMetadatas does not include directories.

## Related Issue
#2176 

## Motivation and Context

To provide support for JuiceFS objectStorage implementation.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: client/daemon/objectstorage/objectstorage.go [hunk 1]
```diff
@@ -388,6 +388,7 @@ func (o *objectStorage) putObject(ctx *gin.Context) {
 		fileHeader  = form.File
 	)
 
+	logger.Infof("put object to %s", objectKey)
```
Comment (gaius-qi) [inline client/daemon/objectstorage/objectstorage.go:391]: Remove log.

File: pkg/objectstorage/oss.go [hunk 2]
```diff
@@ -135,7 +136,7 @@ func (o *oss) GetObjectMetadatas(ctx context.Context, bucketName, prefix, marker
 		return nil, err
 	}
 
-	if limit == 0 {
+	if limit == 0 || limit > DefaultGetObjectMetadatasLimit {
```
Comment (gaius-qi) [inline pkg/objectstorage/oss.go:139]: Remove `|| limit > DefaultGetObjectMetadatasLimit`.

**Review by gaius-qi — CHANGES_REQUESTED** [review]

File: client/daemon/objectstorage/objectstorage.go (hunk 1 above)
Comment (XDTD) [reply client/daemon/objectstorage/objectstorage.go]: done

File: pkg/objectstorage/oss.go (hunk 2 above)
Comment (XDTD) [reply pkg/objectstorage/oss.go]: done

**Review by gaius-qi — CHANGES_REQUESTED** [review]
Add default httpclient.

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**